### ISSUE TITLE
Can't use ORDER BY @rid DESC in distributed mode

### ISSUE THREAD

**Issue by tobiemh** [orientechnologies/orientdb#3750]
Setup the database:

```sql
CONNECT remote:127.0.0.1/testdatabase root root
CREATE CLASS Person extends V
CREATE VERTEX Person SET fullname = 'Joe Bloggs', archived = false
CREATE VERTEX Person SET fullname = 'Tom Jones', archived = false
CREATE VERTEX Person SET fullname = 'Mark Zuckerberg', archived = false
```

Console output:
```shell
Created vertex 'Person#30:0{fullname:Joe Bloggs,archived:false} v1' in 0.051000 sec(s).
Created vertex 'Person#30:1{fullname:Tom Jones,archived:false} v1' in 0.040000 sec(s).
Created vertex 'Person#30:2{fullname:Mark Zuckerberg,archived:false} v1' in 0.046000 sec(s)
```

When executing the following queries, an error is raised:

```sql
-- Ordered by @rid descending
SELECT FROM Person ORDER BY @rid DESC
-- Ordered by @rid descending AND with @rid in WHERE clause
SELECT FROM Person WHERE ( @rid = #30:0 OR archived = false ) ORDER BY @rid DESC
```

These queries work correctly:
```sql
-- Ordered by @rid ascending
SELECT FROM Person ORDER BY @rid ASC
-- Ordered by @rid ascending AND with @rid in WHERE clause
SELECT FROM Person WHERE ( @rid = #30:0 OR archived = false ) ORDER BY @rid ASC
-- Ordered by @rid descending but selecting from a cluster which has content
SELECT FROM cluster:30 WHERE ( @rid = #30:0 OR archived = false ) ORDER BY @rid DESC
-- Ordered by @rid descending but selecting from a cluster which does not have content
SELECT FROM cluster:31 WHERE ( @rid = #30:0 OR archived = false ) ORDER BY @rid DESC
SELECT FROM cluster:32 WHERE ( @rid = #30:0 OR archived = false ) ORDER BY @rid DESC
```

This is the full stack trace raised on the server:

```shell
2015-03-11 02:53:45:810 SEVERE Internal server error:
com.orientechnologies.common.io.OIOException: You cannot access outside the file size (0 bytes). You have requested portion -65536-0 bytes. File: File: person_core-5.cpm os-size=1024, stored=0, filled=0, max=0 [ONetworkProtocolHttpDb]
```

***

OrientDB version: 2.0.4 and 2.0.5-SNAPSHOT

**Comment by SpareShade** [comment]
I am getting the same error in using 2.0.3 in non distributed mode.
```com.orientechnologies.common.io.OIOException: You cannot access outside the file size (0 bytes). You have requested portion -65536-0 bytes. File: File: artworkloan_1.cpm os-size=1024, stored=0, filled=0, max=0```
The strange thing is that the error appears on a production server, and not on a development one.

**Comment by lvca** [comment]
Have you already tried to upgrade to 2.0.8?

**Comment by lvca** [comment]
This should be fixed in 2.1-rc5. Any chance to try on that release?

**Comment by lvca** [comment]
Closed because no answer after a while. In case please comment or reopen this.
